src/runmode-dpdk.c

```diff
@@ -0,0 +1,1274 @@
+                if (iconf->pkt_mempool_array[queue_id] != NULL) {
+                    rte_mempool_free(iconf->pkt_mempool_array[queue_id]);
+                }
+            }
+            SCFree(iconf->pkt_mempool_array);
+        }
+
+        SCFree(iconf);
+    }
+}
+
+void ConfigInit(DPDKIfaceConfig **iconf)
+{
+    DPDKIfaceConfig *ptr = NULL;
+    ptr = SCCalloc(1, sizeof(DPDKIfaceConfig));
+    if (ptr == NULL) {
+        SCLogError(SC_ERR_DPDK_CONF, "Could not allocate memory for DPDKIfaceConfig");
+        SCReturn;
+    }
+
+    ptr->pkt_mempool_array = NULL;
+    ptr->out_port_id = -1; // make sure no port is set
+    SC_ATOMIC_INIT(ptr->ref);
+    (void)SC_ATOMIC_ADD(ptr->ref, 1);
+    ptr->DerefFunc = DPDKDerefConfig;
+    ptr->flags = 0;
+
+    *iconf = ptr;
+}
+
+ConfNode *ConfSetRootIfaceNode(const char *dpdk_node_name, const char *iface)
+{
+    ConfNode *if_root;
+    ConfNode *dpdk_node;
+    /* Find initial node */
+    dpdk_node = ConfGetNode(dpdk_node_name);
+    if (dpdk_node == NULL) {
+        SCLogWarning(SC_WARN_DPDK_CONF, "unable to find %s config", dpdk_node_name);
+    }
+
+    if_root = ConfFindDeviceConfig(dpdk_node, iface);
+    if (if_root == NULL) {
+        SCLogWarning(SC_WARN_DPDK_CONF, "unable to find interface %s in DPDK config", iface);
+    }
+
+    return if_root;
+}
+
+ConfNode *ConfSetDefaultNode(const char *dpdk_node_name)
+{
+    ConfNode *if_default;
+    ConfNode *dpdk_node;
+    /* Find initial node */
+    dpdk_node = ConfGetNode(dpdk_node_name);
+    if (dpdk_node == NULL) {
+        SCLogWarning(SC_WARN_DPDK_CONF, "unable to find %s config", dpdk_node_name);
+    }
+
+    if_default = ConfFindDeviceConfig(dpdk_node, "default");
+    if (if_default == NULL) {
+        SCLogWarning(SC_WARN_DPDK_CONF, "unable to find default interface in DPDK config");
+    }
+
+    return if_default;
+}
+
+int ConfSetRootAndDefaultNodes(
+        const char *dpdk_node_name, const char *iface, ConfNode **if_root, ConfNode **if_default)
+{
+    *if_root = ConfSetRootIfaceNode(dpdk_node_name, iface);
+    *if_default = ConfSetDefaultNode(dpdk_node_name);
+
+    if (*if_root == NULL && *if_default == NULL) {
+        SCLogError(SC_ERR_DPDK_CONF,
+                "unable to find DPDK config for "
+                "interface \"%s\" or \"default\", using default values",
+                iface);
+        SCReturnInt(-ENODEV);
+    }
+
+    /* If there is no setting for current interface use default one as main iface */
+    if (*if_root == NULL) {
+        *if_root = *if_default;
+        *if_default = NULL;
+    }
+
+    SCReturnInt(0);
+}
+
+int ConfigSetIface(DPDKIfaceConfig *iconf, const char *entry_str)
+{
+    int retval;
+
+    if (entry_str == NULL || entry_str[0] == '\0') {
+        SCLogError(SC_ERR_INVALID_VALUE, "Passed value is NULL");
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+
+    retval = rte_eth_dev_get_port_by_name(entry_str, &iconf->port_id);
+    if (retval < 0) {
```

Comment:
as mentioned in comments for DPDK YAML, please rework to identify DPDK port id for copy interface and populate information.